PointerStar/Client/ViewModels/RoomViewModel.cs

```diff
@@ -69,14 +74,36 @@ public bool IsTeamMember
         }
     }
 
-    public RoomViewModel(IRoomHubConnection roomHubConnection, ICookie cookie, HttpClient httpClient)
+    public RoomViewModel(IRoomHubConnection roomHubConnection, ICookie cookie, HttpClient httpClient, IClipboardService clipboardService)
     {
         RoomHubConnection = roomHubConnection ?? throw new ArgumentNullException(nameof(roomHubConnection));
         Cookie = cookie ?? throw new ArgumentNullException(nameof(cookie));
         HttpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+        ClipboardService = clipboardService ?? throw new ArgumentNullException(nameof(clipboardService));
+
         RoomHubConnection.RoomStateUpdated += RoomStateUpdated;
     }
 
+    public async Task OnClickClipboard(object sender, string? url)
```

Comment:
```suggestion
    public async Task OnClickClipboard(string? url)
```

Reply:
I left this here because evetually when I figure out how to send the button reference as a sender, we can onfocus the button once the the copytoclipboard has finished. right now it stays focused. also we wouldn't need the properties CopyButtonText and CopyButtonIcon, because we could just access the button's properties and change it directly.

I couldn't figure out how send the butto reference. Sending the `this` sends the page reference. using `httpelementbutton,getelementbyid` didn't work either. just errors saying it's not available :-/

Reply:
The issue there is we probably don't want to do that. In general you don't want to pass view objects, such as buttons, into the view model.

Reply:
![image](https://user-images.githubusercontent.com/16536254/210185334-c80a3cfc-3251-4c7e-bdce-a51d739c43ef.png)

i didn't mean so much as passing, but usually, in an eventhandler method the sender parameter is the UIElement that called that handler. here I never got it to work, finally I tried sending "this", but I got the page insteasd of the button.